perfkitbenchmarker/linux_benchmarks/cloudsuite_dataserving_benchmark.py

```diff
@@ -0,0 +1,219 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the dataserving benchmark of Cloudsuite.
+
+More info: http://cloudsuite.ch/dataserving/
+"""
+
+import re
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'cloudsuite_dataserving'
+BENCHMARK_CONFIG = """
+cloudsuite_dataserving:
+  description: >
+      Run YCSB against Cassandra. Specify the
+      Cassandra cluster size with --num_vms. Specify the number
+      of YCSB VMs with --ycsb_client_vms.
+  vm_groups:
+    servers:
+      vm_spec: *default_single_core
+      vm_count: 1
+    clients:
+      vm_spec: *default_single_core
+      vm_count: 1
+"""
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  return config
+
+
+def _HasDocker(vm):
+  resp, _ = vm.RemoteCommand('command -v docker',
+                             ignore_failure=True,
+                             suppress_warning=True)
+  if resp.rstrip() == "":
+    return False
+  else:
+    return True
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub.
+        Create a table and load the dataset.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  servers = benchmark_spec.vm_groups['servers']
+  clients = benchmark_spec.vm_groups['clients']
+
+  for vm in servers:
+    if not _HasDocker(vm):
+      vm.Install('docker')
+    vm.RemoteCommand('sudo docker pull cloudsuite/data-serving:server')
+    vm.RemoteCommand('sudo docker run -d --name cassandra-server --net host '
+                     'cloudsuite/data-serving:server')
+
+  for vm in clients:
+    if not _HasDocker(vm):
+      vm.Install('docker')
+    vm.RemoteCommand('sudo docker pull cloudsuite/data-serving:client')
+
+
+def Run(benchmark_spec):
+  """Run the dataserving benchmark.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  client = benchmark_spec.vm_groups['clients'][0]
+  server = benchmark_spec.vm_groups['servers'][0]
+  results = []
+
+  benchmark_cmd = ('sudo docker run --rm --name cassandra-client --net host '
+                   'cloudsuite/data-serving:client %s' % (server.internal_ip))
+  stdout, _ = client.RemoteCommand(benchmark_cmd, should_log=True)
+
+  overall_execution_time = re.findall('\[OVERALL\], RunTime\(ms\), (\d+.?\d*)',
+                                      stdout)[0]
+  results.append(sample.Sample('OVERALL RunTime',
+                               float(overall_execution_time), 'ms'))
```

Comment:
This coding pattern of findall and then append is repeated several times. I suggest a helper function that's called in a loop. Also, consider verifying each pattern is found exactly once. See as example: https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_benchmarks/cloudsuite_in_memory_analytics_benchmark.py#L118